File: 202303Stellargram/backend/observesearch/src/main/java/com/instargram101/observesearch/entity/ObserveSite.java
package com.instargram101.observesearch.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.Comparator;

@Entity
@Table(name = "observe_site")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ObserveSite{
    @Id
    @Column(unique = true, name = "observe_site_id", length = 20, nullable = false)
    private String observeSiteId;

    @Column(length = 20)
    private String name;

    @Column
    private Double latitude;

    @Column
    private Double longitude;

    @ColumnDefault("0")
    private Long reviewCount;

    @ColumnDefault("0")
    private Long ratingSum;

    @Column
    private Long memberId;

}

File: 202303Stellargram/frontend/app/src/main/java/com/ssafy/stellargram/model/ObserveSite.kt
package com.ssafy.stellargram.model

data class SiteInfo(
    val latitude: Double,
    val longitude: Double,
    val name: String,
    val reviewCount: Int,
    val ratingSum: Int
)

data class SiteInfoResponse(
    val code: Int,
    val message: String,
    val data: SiteInfo
)

data class SiteInfoByIdResponse(
    val code: Int,
    val message: String,
    val data: SiteInfoById
)

data class SiteInfoById(
    val observeSite: String,
    val name:String,
    val latitude: Double,
    val longitude: Double,
    val reviewCount: Int,
    val ratingSum: Int,
    val memberId: Long
)